src/coreclr/jit/optcse.cpp

```diff
@@ -785,10 +785,12 @@ bool Compiler::optValnumCSE_Locate()
                 }
 
                 // Don't allow CSE of constants if it is disabled
-                //
                 if (tree->IsIntegralConst())
                 {
-                    if (!enableConstCSE)
+                    if (!enableConstCSE &&
+                        // Unconditionally allow these constant handles to be CSE'd
+                        !tree->IsIconHandle(GTF_ICON_STATIC_HDL) && !tree->IsIconHandle(GTF_ICON_CLASS_HDL) &&
+                        !tree->IsIconHandle(GTF_ICON_STR_HDL))
```

Comment:
Why only these types of handles?

Reply:
> Why only these types of handles?

when I enable all of them I see a lot of size/perfscore regression. I was only interested in GTF_ICON_CLASS_HDL (for PGO), String literals and statics were added because they produced good diffs as well